### ISSUE TITLE
solar_pv_profile_2 should be revised?

### ISSUE THREAD

**Issue by michieldenhaan** [quintel/etsource#2277]
The `solar_pv_profile_2` curve has non-zero loads for every hour of the year. As a result, a scenario using this profile has at least a little bit of PV production in every hour of the year. Since we use this profile for the full load hour slider for solar PV, every (Dutch) scenario with more than 867 FLHs of solar encounters this issue.

![solar_pv_profile_2 csv 2020-05-08 14-39-52](https://user-images.githubusercontent.com/32056448/81406411-d02d1400-9139-11ea-8cd8-023c32f9fc09.png)

@ChaelKruip, @DorinevanderVlies do you know the background of this profile?

**Comment by redekok** [comment]
I guess the problem lies within the [source data](https://data.open-power-system-data.org/time_series/). "Actual solar generation in Spain in MW" is used, provided by the ENTSO-E. As you can see below, according to the source data there are indeed non-zero loads for every hour of the year, which is pretty strange.

![image](https://user-images.githubusercontent.com/5946498/81544969-7cf4d480-9378-11ea-8142-d7c999d9a1b9.png)

We _could_ post-process this data ourselves. If we have data available about the times of sunrise and sunset per day, we could overwrite the nightly hours with zero solar production. This is kind of a hassle though..

**Comment by ChaelKruip** [comment]
> I guess the problem lies within the source data. "Actual solar generation in Spain in MW" is used, provided by the ENTSO-E. As you can see below, according to the source data there are indeed non-zero loads for every hour of the year, which is pretty strange.

I would be very interested in finding out why this is!

> We could post-process this data ourselves. If we have data available about the times of sunrise and sunset per day, we could overwrite the nightly hours with zero solar production. This is kind of a hassle though..

I am not in favour of this. We don't know the assumptions underlying the data so 'processing' it will destroy information in a way without having the full picture.

**Comment by redekok** [comment]
> I would be very interested in finding out why this is!

I could try emailing them about this.

> I am not in favour of this. We don't know the assumptions underlying the data so 'processing' it will destroy information in a way without having the full picture.

Agreed, not a fan of this either.

**Comment by redekok** [comment]
It might be due to some residual solar thermal power generation (didn't realise the time series included both solar pv and solar thermal). Still awaiting the answer of the official contact person though.

**Comment by redekok** [comment]
The contact person of the OPSD time series responded that the non-zero values by night are indeed due to [concentrated solar power](https://en.wikipedia.org/wiki/Solar_power_in_Spain#Solar_thermal_power_plants).

For Spain this makes sense since there is installed capacity of CSP. For regions where no CSP is installed (such as NL) it feels a bit strange to use a CSP-inclusive profile to perform our FLH-interpolation trick.

Maybe we could slightly adjust the interpolation method? And only perform an interpolation for non-zero values? So that the zero values (during nightly hours) remain zero? Or something similar?

@DorinevanderVlies @ChaelKruip what are your thoughts on this?

**Comment by DorinevanderVlies** [comment]
> Maybe we could slightly adjust the interpolation method? And only perform an interpolation for non-zero values? So that the zero values (during nightly hours) remain zero? Or something similar?

Sounds good!

**Comment by ChaelKruip** [comment]
> Maybe we could slightly adjust the interpolation method? And only perform an interpolation for non-zero values? So that the zero values (during nightly hours) remain zero? Or something similar?

I think this would be a last-resort type of fix for me. it might be challenging but I would prefer to go back to the source of the data. If we could separate the CSP from the PV contributions, the whole profile would improve in realism. As a bonus, we might be able to use a dedicated CSP profile for CSP :)

**Comment by michieldenhaan** [comment]
Alternatively we could generate a profile for NL based weather station De Kooy in 2003 (the sunniest year on record). Using our script this would give a profile with 1102 FLHs. That's lower than the current slider max (1361) but still quite 'extreme' given the long year average of 870 FLHs. An advantage of this is that it's based on NL data so sunrise/sunset times are aligned with `solar_pv_profile_1` (in Summer there's a 90 minute difference in sunrise/sunset times between Amsterdam and Madrid).

That would be a relatively easy fix for NL but not for other countries.

**Comment by redekok** [comment]
> If we could separate the CSP from the PV contributions, the whole profile would improve in realism.

Yes, I fully agree. I've already asked whether it's possible for OPSD to provide the separate profiles but haven't heard anything back about this yet.

> Alternatively we could generate a profile for NL based weather station De Kooy in 2003 (the sunniest year on record).

And I also like this fix for NL! I guess it's better to have a clean fix for NL than having a "schijnzekere" (shamming?) fix, although the first would be less consistent with the approach for other countries.

**Comment by redekok** [comment]
The OPSD contact person [pointed out](https://github.com/Open-Power-System-Data/time_series/issues/26#issuecomment-633975621) that the Spanish TSO actually separates [solar PV](https://www.esios.ree.es/en/analysis/1161?vis=1&start_date=01-01-2015T21%3A00&end_date=31-12-2018T23%3A50&compare_start_date=31-12-2014T21%3A00&groupby=hour) and [solar thermal](https://www.esios.ree.es/en/analysis/1162?vis=1&start_date=01-01-2015T00%3A00&end_date=31-12-2015T23%3A50&compare_start_date=31-12-2014T00%3A00&groupby=hour) profiles, which can be exported as CSV-files.

The interesting thing is that the solar PV profile still has non-zero loads for every hour of the year, although the nightly values do approach values very close to zero. Not sure if this could be explained by some kind of residual solar PV production? 🤔

**Solar PV**:
![image](https://user-images.githubusercontent.com/5946498/82949589-2dcbb800-9fa4-11ea-9a85-9c36bbb6db3e.png)

![image](https://user-images.githubusercontent.com/5946498/82950040-ef82c880-9fa4-11ea-8ac0-57fea4f183c3.png)

**Solar thermal**:
![image](https://user-images.githubusercontent.com/5946498/82949643-3e7c2e00-9fa4-11ea-942e-d98e95b6e08f.png)

![image](https://user-images.githubusercontent.com/5946498/82949983-d712ae00-9fa4-11ea-9d73-1810e296c573.png)

**Comment by redekok** [comment]
I've e-mailed the Spanish TSO REE about this. Awaiting their response.

**Comment by redekok** [comment]
They're quick!

"Good morning, it seems that some photovoltaic installations located near areas with artificial lighting (such as motorways) may have a small production of electricity at night."

That explains. Didn't know artificial lighting could be that bright. Anyways, should we separate the PV and CSP profiles (measured generation by REE) for Spain?

**Comment by ChaelKruip** [comment]
> Anyways, should we separate the PV and CSP profiles (measured generation by REE) for Spain?

I would be in favour of that if possible and if we have the underlying data to do it nicely. Curious about the opinion of @AlexanderWirtz on this as well.

**Comment by michieldenhaan** [comment]
We could also use the ENTSO-E PECD dataset. This contains time series for all EU countries for 1982-2016: https://www.entsoe.eu/outlooks/seasonal/
Their data for Spain does not seem to have the issues described here.

**Comment by AlexanderWirtz** [comment]
>Anyways, should we separate the PV and CSP profiles (measured generation by REE) for Spain?

>I would be in favour of that if possible and if we have the underlying data to do it nicely. Curious about the opinion of @AlexanderWirtz on this as well.

I am strongly in favor of separating. The technologies are quite different. CSP stores energy in molten salt, so it keeps producing for hours after sundown. I had no idea it keeps producing until sunrise, though.

> We could also use the ENTSO-E PECD dataset. This contains time series for all EU countries for 1982-2016: https://www.entsoe.eu/outlooks/seasonal/
> Their data for Spain does not seem to have the issues described here.

I am curious how it compares to the data from REE.

**Comment by kndehaan** [comment]
The use of this profile has been retired with the solar PV project #3351. Closing this issue.
